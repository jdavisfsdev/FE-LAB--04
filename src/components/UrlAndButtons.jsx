/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import UrlAndButtonsCSS from './UrlAndButtons.module.css';

export default function UrlAndButtons({ url, onChange, onSubmit }) {
  return (
    <form onSubmit={onSubmit}>
      <input className={UrlAndButtonsCSS.url} data-testid="url" name="url" id="url" placeholder="URL" type="text" onChange={onChange} value={url}/>
      <div className={UrlAndButtonsCSS.container}>
        <label htmlFor="radio" className={UrlAndButtonsCSS.label}>
        Get
          <input onChange={onChange} className={UrlAndButtonsCSS.radio} value={'GET'} name="method" data-testid="get" type="radio" id="get"></input>
        </label>
        <label htmlFor="radio" className={UrlAndButtonsCSS.label}>
        Post
          <input onChange={onChange} className={UrlAndButtonsCSS.radio} value={'POST'} name="method" type="radio" id="post"></input>
        </label>
        <label htmlFor="radio" className={UrlAndButtonsCSS.label}>
        Put
          <input onChange={onChange} className={UrlAndButtonsCSS.radio} value={'PUT'} name="method" type="radio" id="put"></input>
        </label>
        <label htmlFor="radio" className={UrlAndButtonsCSS.label}>
        Patch
          <input onChange={onChange} className={UrlAndButtonsCSS.radio} value={'PATCH'} name="method" type="radio" id="patch"></input>
        </label>
        <label htmlFor="radio" className={UrlAndButtonsCSS.label}>
        Delete
          <input onChange={onChange} className={UrlAndButtonsCSS.radio} value={'DELETE'} name="method" type="radio" id="delete"></input>
        </label>
        <button onSubmit={onSubmit} data-testid="enter" className={UrlAndButtonsCSS.radio}>Go</button>
      </div>
    </form>
  );
}

UrlAndButtons.propTypes = {
    
};
