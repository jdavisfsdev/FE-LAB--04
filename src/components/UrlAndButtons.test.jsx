import React from 'react';
import { render } from '@testing-library/react';
import UrlAndButtons from './UrlAndButtons';


describe('Search component snapshot', () => {
  it('Renders the URL input and radios component on the page', () => {
    const { asFragment } = render(
      <UrlAndButtons />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
