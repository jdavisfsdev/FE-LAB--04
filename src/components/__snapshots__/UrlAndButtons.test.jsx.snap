// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component snapshot Renders a search component on the page 1`] = `
<DocumentFragment>
  <form>
    <input
      class="url"
      name="url"
      placeholder="URL"
      type="text"
      value=""
    />
    <div
      class="container"
    >
      <label
        class="label"
        for="radio"
      >
        Get
        <input
          class="radio"
          id="get"
          name="method"
          type="radio"
          value="GET"
        />
      </label>
      <label
        class="label"
        for="radio"
      >
        Post
        <input
          class="radio"
          id="post"
          name="method"
          type="radio"
          value="POST"
        />
      </label>
      <label
        class="label"
        for="radio"
      >
        Put
        <input
          class="radio"
          id="put"
          name="method"
          type="radio"
          value="PUT"
        />
      </label>
      <label
        class="label"
        for="radio"
      >
        Patch
        <input
          class="radio"
          id="patch"
          name="method"
          type="radio"
          value="PATCH"
        />
      </label>
      <label
        class="label"
        for="radio"
      >
        Delete
        <input
          class="radio"
          id="delete"
          name="method"
          type="radio"
          value="DELETE"
        />
      </label>
      <button
        class="radio"
      >
        Go
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`Search component snapshot Renders the URL input and radios component on the page 1`] = `
<DocumentFragment>
  <form>
    <input
      class="url"
      name="url"
      placeholder="URL"
      type="text"
      value=""
    />
    <div
      class="container"
    >
      <label
        class="label"
        for="radio"
      >
        Get
        <input
          class="radio"
          id="get"
          name="method"
          type="radio"
          value="GET"
        />
      </label>
      <label
        class="label"
        for="radio"
      >
        Post
        <input
          class="radio"
          id="post"
          name="method"
          type="radio"
          value="POST"
        />
      </label>
      <label
        class="label"
        for="radio"
      >
        Put
        <input
          class="radio"
          id="put"
          name="method"
          type="radio"
          value="PUT"
        />
      </label>
      <label
        class="label"
        for="radio"
      >
        Patch
        <input
          class="radio"
          id="patch"
          name="method"
          type="radio"
          value="PATCH"
        />
      </label>
      <label
        class="label"
        for="radio"
      >
        Delete
        <input
          class="radio"
          id="delete"
          name="method"
          type="radio"
          value="DELETE"
        />
      </label>
      <button
        class="radio"
      >
        Go
      </button>
    </div>
  </form>
</DocumentFragment>
`;
